test/node-api.js

```diff
@@ -6,11 +6,188 @@ chai.use(codeEquality);
 
 const { expect } = chai;
 
-const p = new Preprocessor();
+describe("parse", function() {
+  it("basic example", function() {
+    let p = new Preprocessor();
+    let output = p.parse("<template>Hello!</template>");
 
-describe("something", function () {
-  it("works for a basic example", function () {
-    let output = p.process("<template>Hi</template>");
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 0,
+          end: 27
+        },
+        contentRange: {
+          start: 10,
+          end: 16
+        },
+        startRange: {
+          end: 10,
+          start: 0
+        },
+        endRange: {
+          start: 16,
+          end: 27
+        }
+      },
+    ]);
+  });
+
+  it("expression position", function() {
+    let p = new Preprocessor();
+    let output = p.parse("const tpl = <template>Hello!</template>");
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 12,
+          end: 39
+        },
+        contentRange: {
+          start: 22,
+          end: 28
+        },
+        startRange: {
+          start: 12,
+          end: 22
+        },
+        endRange: {
+          start: 28,
+          end: 39
+        }
+      },
+    ]);
+  });
+
+  it("inside class body", function() {
+    let p = new Preprocessor();
+    let output = p.parse(`
+      class A {
+        <template>Hello!</template>
+      }
+    `);
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 25,
+          end: 52
+        },
+        contentRange: {
+          start: 35,
+          end: 41
+        },
+        startRange: {
+          start: 25,
+          end: 35
+        },
+        endRange: {
+          start: 41,
+          end: 52
+        }
+      },
+    ]);
+  });
+
+  it("preceded by a slash character", function() {
+    let p = new Preprocessor();
+    // What is this testing?
```

Comment:
This kind of test is fairly pointless here. It's a reproduction for bad parsing behavior in the earlier implementation. But that didn't have a real javascript parser to rely on, and we do.

Reply:
I agree, do we want to delete it?

Alternatively, I think this is a somewhat novel test case along these lines that is perhaps worth having: `const Foo = 1 / <template>...</template>;`. It's meaningless (runtime type error) for `<template>` tag, but it's a legal expression position and perhaps could be useful for, hypothetically `const Result = 1 / <formula>...</formula>`.

Reply:
They're inexpensive to keep, I only commented because of the comment in the code.